Add image scan script

ivanvc opened the pull request:
Introduce a new script that can run on stable branches that checks the vulnerabilities for the latest tag (from the given branch).

Adds Trivy as a tool dependency, to keep track of the latest released version.

A periodic Prow job will later execute this. We'll get alerts when a CVE is found in our images, either by a direct vulnerability or for a dependency with a reported vulnerability of high or critical severity.

Part of #19363.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

ivanvc commented:
Oof, inconsistent dependencies. I'll draft and will undraft once I address those :sweat_smile:.

ivanvc commented:
/test pull-etcd-verify

ivanvc commented:
Adding Trivy as a tool bumped many indirect dependencies, resulting in changes to many `go.mod`s.

ahrtr commented:
My understanding is that you only introduce a new dependency on `github.com/aquasecurity/trivy`, could you move unrelated dependencies bumping into a separate PR to make this PR easier to review?

ivanvc commented:
> My understanding is that you only introduce a new dependency on `github.com/aquasecurity/trivy`, could you move unrelated dependencies bumping into a separate PR to make this PR easier to review?

I opened https://github.com/etcd-io/etcd/pull/19819.

But I wonder if it makes more sense to open a pull request to **only bump** the existing dependencies **without introducing Trivy**. Let me know, and I can do that instead.

ahrtr commented:
@ivanvc Probably I did not say it clearly in my previous comment. 

Adding dependency `github.com/aquasecurity/trivy` is OK, because it's required for the image scan script. It's OK to do both things (see below) in one PR,
- Add dependency `github.com/aquasecurity/trivy`
- Add image scan script

But I see that you bumped many other dependencies as well in this PR. Can you move all other dependencies bumping into a separate PR?

ivanvc commented:
@ahrtr, then, please take a look at #19821.

I'll close #19819, then.

ahrtr commented:
Please rebase this PR.

ivanvc commented:
I'll get back to this shortly. I'm a bit busy and reconsidering how we'll implement this in the stable release branches. Either way, the trivy dependency is required, so it's good that we bumped the indirect dependencies.

ivanvc committed:
Add trivy tool through Dockerfile

Even though Trivy is a Go project, it's not possible to track it as our
other tools (`tools/mod/tools.go`), because they point the Go directive
to the latest version patch. This breaks our Go workspace.

Signed-off-by: Ivan Valdes <ivan@vald.es>

ivanvc committed:
Add Trivy image scanning script

Signed-off-by: Ivan Valdes <ivan@vald.es>
